workflow-runtime/src/main/java/com/squareup/workflow1/RenderingAndSnapshot.kt

```diff
@@ -21,4 +21,21 @@ package com.squareup.workflow1
 data class RenderingAndSnapshot<out RenderingT>(
   val rendering: RenderingT,
   val snapshot: TreeSnapshot
-)
+) {
+
+  override fun equals(other: Any?): Boolean {
+    return when {
+      this === other -> true
+      javaClass != other?.javaClass -> false
+      else -> rendering == (other as RenderingAndSnapshot<*>).rendering
```

Comment:
We could still do an identity comparison for `snapshot`, but then we'd need to use `System.identityHashCode` for the hashCode implementation and I don't believe there's any multiplatform version of that function (although there is a [native one](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.native/identity-hash-code.html), apparently). Anyway, probably more work than it's worth.

Although I'm kind of thinking it would be better to just make this a simple identity comparison and not include even the rendering, since it seems overly complicated, and not useful, to have only partial equality.

Reply:
> Although I'm kind of thinking it would be better to just make this a simple identity comparison and not include even the rendering, since it seems overly complicated, and not useful, to have only partial equality.

Yeah, I like that better, seems less magical. And that could be as simple as dropping `data` from the class declaration.